dspy/streaming/streaming_listener.py

```diff
@@ -217,10 +214,13 @@ def flush(self) -> str:
                 f"{', '.join([a.__name__ for a in ADAPTER_SUPPORT_STREAMING])}"
             )
 
-    def _signature_field_is_citation_type(self) -> bool:
-        """Check if the signature field is a citations field."""
-        from dspy.predict import Predict
-        return isinstance(self.predict, Predict) and getattr(self.predict.signature.output_fields.get(self.signature_field_name, None), "annotation", None) == Citations
+    @property
+    def output_type(self) -> type | None:
+        try:
+            return self.predict.signature.output_fields[self.signature_field_name].annotation
+        except Exception:
```

Comment:
when will this throw an exception?

Reply:
It does not fail in normal cases, but since `predict` or `signature_field_name` is not guaranteed to be non-null, it's better to have a safeguard

Reply:
I know, but typing-wise they are nullable (https://github.com/stanfordnlp/dspy/blob/28a7f786f890d8bcfd9aa83bedf1465c97deb940/dspy/streaming/streaming_listener.py#L24-L29), so I still want to keep the check. We can do if chain, as well.